conda/cli/install.py

```diff
@@ -265,15 +267,22 @@ def install(args, parser, command='install'):
             )
 
     except ResolvePackageNotFound as e:
-        channels_urls = tuple(calculate_channel_urls(
-            channel_urls=index_args['channel_urls'],
-            prepend=index_args['prepend'],
-            platform=None,
-            use_local=index_args['use_local'],
-        ))
-        raise PackagesNotFoundError(e._formatted_chains, channels_urls)
+        if args.repodata_fn != 'repodata.json':
+            args.repodata_fn = 'repodata.json'
+            return install(args, parser, command)
```

Comment:
I have 2 questions:
1. Is there a situation when someone might want to fall back on another repodata other than repodata.json? For example, if the first repodata doesn't work, then use a slightly larger view of repodata, then finally fall back on repodata.json.
2. How often is it expected for the package to not be found? In that case solving takes twice as long. Is ensuring that this doesn't happen super often a responsibility of the bit that generates repodata?

Reply:
1. Users may want to use an epoch based repodata if the repodata with the lastest releases does not resolve.  One could imagine a system with multiple layers of fall backs. Each attempt to solve the environment with a particular repodata will have a time cost.  A preferred sparse repodata followed by a single fallback seems like a good initial step before implementing a fully layered approach.
2. Each solve will take time and if a solution is not found in the preferred repodata file that is time wasted.  The preferred repodata file default to a version with just the latest version of all packages which should be small and quick to process and check for a solution. The conda solver prefers these version when installing unbound specification so I expect that solutions will be found regularly via this method.  Basically, simple solves which use the latest version should be very quick (<5 seconds).  More complex solves which require the fallback repodata and second or two overhead to check the "easy" solution should not be a significant overhead.

Reply:
gotcha! For (2) it would be interesting to understand how many solves it needs for most users. Maybe we can look at hits on all the repodata.json's?